src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java

```diff
@@ -6164,6 +6164,25 @@ && isInternalCase(asylumCase)
         );
     }
 
+    @Bean
+    public PreSubmitCallbackHandler<AsylumCase> internalUpdateTribunalDecisionRule31NotificationHandler(
+        @Qualifier("internalUpdateTribunalDecisionRule31LetterNotificationGenerator")
+        List<NotificationGenerator> notificationGenerators) {
+
+        return new NotificationHandler(
+            (callbackStage, callback) -> {
+                AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+                return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                    && callback.getEvent() == Event.UPDATE_TRIBUNAL_DECISION
+                    && isInternalCase(asylumCase)
+                    && isRule31ReasonUpdatingDecision(asylumCase);
```

Comment:
We've also been adding a check for:
- && !isAppellantInDetention(asylumCase)
- && hasAppellantAddressInCountryOrOutOfCountry(asylumCase);